src/Analyzer.Cli.UnitTests/TemplateDiscoveryTests.cs

```diff
@@ -0,0 +1,88 @@
+        {
+            var discoveredTemplates = TemplateDiscovery.DiscoverTemplatesAndParametersInDirectory(new DirectoryInfo(testTemplateDirectory));
+
+            Assert.AreEqual(18, discoveredTemplates.Count());
+
+            // Create list of templates expecting a match with parameters
+            var subDirectory = "ToTestSeparateParametersFile";
+            var templateName = "TemplateWithSeparateParametersFile";
+
+            var templatesWithParameters = new[] {
+                Path.Combine(testTemplateDirectory, subDirectory, templateName + ".bicep"),
+                Path.Combine(testTemplateDirectory, subDirectory, templateName + ".json") };
+
+            var parametersToMatchToTemplates = new[] {
+                Path.Combine(testTemplateDirectory, subDirectory, templateName + ".parameters.json"),
+                Path.Combine(testTemplateDirectory, subDirectory, templateName + ".parameters-dev.json") };
+
+            List<TemplateAndParams> pairsToAssertWereFound = new();
+
+            foreach (var template in templatesWithParameters)
+                foreach (var parameters in parametersToMatchToTemplates)
+                    pairsToAssertWereFound.Add(new(new FileInfo(template), new FileInfo(parameters)));
+
+            // Verify parameter files discovered with templates
+            foreach (var pair in pairsToAssertWereFound)
+            {
+                Assert.IsNotNull(discoveredTemplates.Single(t => t.Template.FullName == pair.Template.FullName && t.Parameters.FullName == pair.Parameters.FullName),
+                    $"Expected template {pair.Template.FullName} and parameters file {pair.Parameters.FullName} not discovered.");
+            }
+        }
+
+        [DataTestMethod]
+        [DataRow("AppServicesLogs-Passes.json", DisplayName = "ARM JSON template with no matching parameters")]
+        [DataRow("AppServicesLogs-Passes.bicep", DisplayName = "Bicep template with no matching parameters")]
+        [DataRow(@"ToTestSeparateParametersFile\TemplateWithSeparateParametersFile.json",
+            "TemplateWithSeparateParametersFile.parameters.json",
+            "TemplateWithSeparateParametersFile.parameters-dev.json",
+            DisplayName = "ARM JSON template with 2 matching parameters files")]
+        [DataRow(@"ToTestSeparateParametersFile\TemplateWithSeparateParametersFile.bicep",
+            "TemplateWithSeparateParametersFile.parameters.json",
+            "TemplateWithSeparateParametersFile.parameters-dev.json",
+            DisplayName = "Bicep template with 2 matching parameters files")]
+        public void FindParameterFilesForTemplate_ValidTemplateFile_ReturnsExpectedParametersFiles(string templateName, params string[] parameterFiles)
+        {
+            var templateNameSplit = templateName.Split('\\');
+            var directoryForTest = Path.Combine(templateNameSplit[..^1].Prepend(testTemplateDirectory).ToArray());
```

Comment:
this is fine, but not sure if it could match the other instances more. (saying this without verifying it fully or thinking too hard).

Reply:
I added some comments to clarify.

Reply:
mkay.